Editor/Windows/AssetPaletteProperties.cs

```diff
@@ -159,20 +153,16 @@ public SerializedProperty SelectedFolderSerializedProperty
                                     || cachedSelectedFolderSerializedProperty.GetValue<PaletteFolder>() == null))
                 {
                     ClearCachedSelectedFolderSerializedProperties();
-                    SelectedFolderReferenceIdPath = null;
                 }
                 
                 if (!didCacheSelectedFolderSerializedProperty)
                 {
                     folderPanel.EnsureFolderExists();
                     didCacheSelectedFolderSerializedProperty = true;
-                    
-                    // First try to find the selected folder by reference id path. Don't use a regular property path
-                    // because those have indices baked into it and those get real screwy when you move things around.
-                    cachedSelectedFolderSerializedProperty =
-                        CurrentCollectionSerializedObject.FindPropertyFromReferenceIdPath(
-                            SelectedFolderReferenceIdPath,
-                            FolderPanel.RootFoldersPropertyName,
+
+
```

Comment:
Would be good to explain why we use this GUID path instead of the serialized property's `propertyPath`, I don't think that's inherently clear.

```suggestion
// First try to find the selected folder by reference guid path. Don't use a regular property path
// because those have indices baked into it and those get real screwy when you move things around.
```